### PR TITLE
Graph UI improvements

### LINKED ISSUE #88: Fix component layering
Currently the control of the graph is impeded by the layering of the components. 
The base layout may need to be adapted to make this work better.

### PR DESCRIPTION
**Pull request by yrahul3910**
- Grab citation counts from DOIs
- Working towards tooltips on nodes
- Results of live coding

This PR is the first organized attempt at improving the graph UI. Specifically, it aims to address #65 and #66. There are some issues in the
current code, which are appropriately documented (see `GraphView.vue`, L148). These commits were a collaboration between @yrahul3910 and
@andre-motta. Future commits to this branch will also attempt to fix #88, which is more pressing.

### CODE REVIEW

File: workflow/wp-ba-results.csv [hunk 1]
```diff
@@ -0,0 +1,16 @@
+gh_id, paper_doi, reuse_type, comment, citation_number, reused_doi, alt_url, page_num
```
Comment (bhermann) [inline workflow/wp-ba-results.csv:1]: Why is this file here? Is this an incomplete result? is it done?

File: department-of-reuse/src/backend/RIndex.ts [hunk 2]
```diff
@@ -22,6 +22,13 @@ export default class RIndex {
             }));
     }
 
+    async computeCitationCountForWork(sourceDoi: string) : Promise<number> {
```
Comment (bhermann) [inline department-of-reuse/src/backend/RIndex.ts:25]: As this function technically has nothing to do with computing the RIndex, you might move it directly into GraphView.vue

**Review by bhermann — CHANGES_REQUESTED** [review]

File: workflow/wp-ba-results.csv (hunk 1 above)
Comment (yrahul3910) [reply workflow/wp-ba-results.csv:1]: I'm not entirely sure. I don't recall having moved/copied that file there.

File: department-of-reuse/src/backend/RIndex.ts (hunk 2 above)
Comment (yrahul3910) [reply department-of-reuse/src/backend/RIndex.ts:25]: Will do!

File: department-of-reuse/src/components/GraphView.vue [hunk 3]
```diff
@@ -39,8 +42,11 @@ export default {
     }
 
     async function createNodeFromDOI(doi : string) : Promise<NodeDefinition> {
+      const indexer = new RIndex(props.reuseData);
+      const citationCount = await indexer.computeCitationCountForWork(doi)
+
       const title = await getItemTitle(doi);
-      return { data: {id: doi, name : title }};
+      return { data: {id: doi, name : title, citations: citationCount }};
```
[the remark this answers is not in the document]
Comment (yrahul3910) [reply department-of-reuse/src/components/GraphView.vue:49]: I'll check that. Before the PR I ran

```
(develop) $ git merge main
(develop) $ gh pr create
```

So the merge should've be okay. It does seem like the line you point has the merge conflict for some reason, so I'll resolve that.

[the remark this answers is not in the document]
Comment (yrahul3910) [reply department-of-reuse/src/components/GraphView.vue]: Made some changes. Sadly, I set `pull.rebase` false, so it shows 70-ish commits when the changes are fewer. Could you please check?

File: workflow/wp-ba-results.csv (hunk 1 above)
Comment (bhermann) [reply workflow/wp-ba-results.csv:1]: It's part of @andre-motta 's "-temp" commit 7d7f981f4f53632744cf3fbac95ae88800eba7ac

Comment (bhermann) [reply workflow/wp-ba-results.csv:1]: This is the only point holding back this PR. Could you remove it for now and stash it until we clarified what this file was for?
